### PR TITLE
chore(connections): remove single connection ui and hooks COMPASS-8469

### PR DESCRIPTION
**Pull request by gribnoysup**
Before I start removing usage of deprecated connections repository and connections manager hooks and locators, I want to clean-up some unused UI that's still using those, otherwise it's hard to see what actually needs refactoring and what will get removed anyway

### CODE REVIEW

File: packages/compass-connections/src/index.tsx [hunk 1]
```diff
@@ -34,6 +33,7 @@ const ConnectionsComponent: React.FunctionComponent<{
   onAutoconnectInfoRequest?: (
     connectionStorage: ConnectionStorage
   ) => Promise<ConnectionInfo | undefined>;
+  allowAutoconnectInfoReconnect?: boolean;
```
Comment (Anemy) [inline packages/compass-connections/src/index.tsx:36]: nit: Maybe it's me, it's still a bit hard to understand what the purpose of this boolean is from the name. Is there another name we could give it? 

Thinking of a name from my understanding, it ends up being super long, so maybe this is fine as is. 
`allowReconnectWhenConnectionInfoHasAutoconnected`?

**Review by Anemy — APPROVED** [review]

File: packages/compass-connections/src/index.tsx (hunk 1 above)
Comment (gribnoysup) [reply packages/compass-connections/src/index.tsx:36]: Yeah, fair, I'll update, your proposal is closer, but still not there a bit (I'm thinking if I flip the value it maybe will be better? `doNotReconnectDisconnectedAutoconnectInfo`, that's like exactly what we're trying to do here I guess even if it's super verbose), it is getting long, but it's a corner case handling thing so probably okay 🙂

File: packages/compass/src/app/components/home.tsx [hunk 2]
```diff
@@ -112,35 +72,13 @@ const verticalSplitStyles = css({
...
-  const appRegistry = useLocalAppRegistry();
-  const { connectionInfo, isConnected, disconnect } =
-    useSingleConnectionModeConnectionInfoStatus();
-
-  useEffect(() => {
-    function onDisconnect() {
-      void disconnect();
-    }
-
-    hadronIpc.ipcRenderer?.on('app:disconnect', onDisconnect);
```
Comment (lerouxb) [inline packages/compass/src/app/components/home.tsx:136]: So we don't have an equivalent feature to this anymore?

File: packages/compass/src/app/components/home.tsx [hunk 3]
```diff
@@ -152,13 +90,6 @@ function Home({
     [showCollectionSubMenu, hideCollectionSubMenu]
   );
 
-  useEffectOnChange(() => {
-    if (!isConnected) {
-      hideCollectionSubMenu();
```
Comment (lerouxb) [inline packages/compass/src/app/components/home.tsx:157]: Same question here - is this not a thing in MC world?

**Review by lerouxb — APPROVED** [review]

File: packages/compass/src/app/components/home.tsx (hunk 3 above)
Comment (gribnoysup) [reply packages/compass/src/app/components/home.tsx:157]: I don't think this meaningfully did anything for quite some time already (to be more specific I think it was only causing bugs in some corner cases where you'r "first connection" would change and just nuke the existing submenu unexpectedly), this is handled through the `onWorkspaceChanged` handler as a side-effect of active workspace changing for all cases

File: packages/compass-connections/src/index.tsx (hunk 1 above)
Comment (lerouxb) [reply packages/compass-connections/src/index.tsx:36]: Inverting it makes sense to me 👍

**Pull request merged**